kernel/src/snapshot.rs

```diff
@@ -287,7 +287,12 @@ struct CheckpointMetadata {
 
 /// Try reading the `_last_checkpoint` file.
 ///
-/// In case the file is not found, `None` is returned.
+/// Note that we typically want to ignore a missing/invalid `_last_checkpoint` file without failing
+/// the read. Thus, the semantics of this function are to return `None` if the file is not found or
+/// is invalid JSON. Unexpected/unrecoverable errors are returned as `Err` case and are assumed to
+/// cause failure.
+///
+/// TODO: java kernel retries three times before failing, should we do the same?
```

Comment:
my initial instinct would be, that if retry made sense, then on io errors when trying to read the file. I would expect this to be the responsibility of of the engine inside `read_files`. This I think is is good as is. OR do we get errors back, that we think are retryable?

Reply:
my initial reaction is also to leave this to the engine. I don't think there is anything specific about reading the _last_checkpoint file and if failures occur across the network it should be handled the same for all our reads and taken care of within `read_files`